src/utils/viewer/viewer.cpp

```diff
@@ -118,56 +108,46 @@ class Viewer : public Mn::Platform::Application {
     }
   }
 
-  esp::assets::ResourceManager resourceManager_;
-  // SceneManager must be before physicsManager_ as the physicsManager_
-  // assumes that it "owns" things owned by the scene manager
-  esp::scene::SceneManager sceneManager_;
+  // The simulator object backend for this viewer instance
+  std::unique_ptr<esp::sim::Simulator> simulator_;
 
-  std::shared_ptr<esp::physics::PhysicsManager> physicsManager_;
+  // The managers belonging to the simulator
+  std::shared_ptr<esp::assets::managers::ObjectAttributesManager>
+      objectAttrManager_ = nullptr;
+  std::shared_ptr<esp::assets::managers::AssetAttributesManager>
+      assetAttrManager_ = nullptr;
+  std::shared_ptr<esp::assets::managers::StageAttributesManager>
+      stageAttrManager_ = nullptr;
+  std::shared_ptr<esp::assets::managers::PhysicsAttributesManager>
+      physAttrManager_ = nullptr;
 
   bool debugBullet_ = false;
 
-  std::vector<int> sceneID_;
+  esp::scene::SceneNode* rootNode_ = nullptr;
   esp::scene::SceneNode* agentBodyNode_ = nullptr;
-  esp::scene::SceneNode* rgbSensorNode_ = nullptr;
-
-  std::string stageFileName;
-  esp::scene::SceneGraph* sceneGraph_;
-  esp::scene::SceneNode* rootNode_;
-
-  int navMeshVisPrimID_ = esp::ID_UNDEFINED;
-  esp::scene::SceneNode* navMeshVisNode_ = nullptr;
 
+  std::string sceneFileName;
```

Comment:
Is this stageFileName or sceneFileName? Maybe add a comment it can also be stage?